src/frontends/pytorch/src/op/dot.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "pt_framework_node.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_dot(const NodeContext& context) {
+    num_inputs_check(context, 2, 3);
+    // "aten::dot(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)"
+
+    auto tensor1 = context.mark_node(std::make_shared<v0::Convert>(context.get_input(0), element::f32));
+    auto tensor2 = context.mark_node(std::make_shared<v0::Convert>(context.get_input(1), element::f32));
+
+    auto multiply = context.mark_node(std::make_shared<v1::Multiply>(tensor1, tensor2));
+    auto axes = context.mark_node(v0::Constant::create(element::i64, Shape{1}, {0}));
```

Comment:
I'm not too sure about this methodology I've adopted. I haven't found many similar pull requests on how to use the second parameter for ReduceSum. In this case, since I need to reduce the axes to 1, I limited myself to creating an auxiliary variable to rely on and call it within the latter

Reply:
just as an idea, you can use matmul operation (dot-product is specific case of matrix multiplication, but where both inputs are 1d vectors)

Reply:
regarding your question about ReduceSum, you can check openvino docs for this [operation](https://docs.openvino.ai/2023.3/openvino_docs_ops_reduction_ReduceSum_1.html).

it has attribute keep_dims that responsible for opposite behaviour (preserve original axes) if it is true, if it is false - dimensions will be reduced